### PR TITLE
Deploy to prod

### PR DESCRIPTION
**Pull request by james-otten**

### CODE REVIEW

File: .github/workflows/helm_lint.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@e6669bcd63d7cb57cb4380c33043eebe5d111992 # v2.6.1
+
+      - name: Run chart-testing (list-changed)
+        id: list-changed
+        run: |
+          changed=$(ct list-changed --target-branch ${{ github.event.repository.default_branch }})
+          if [[ -n "$changed" ]]; then
+            echo "changed=true" >> "$GITHUB_OUTPUT"
+          fi
```
Comment (WillNilges) [inline .github/workflows/helm_lint.yaml:35]: Question: Is this easily accessible/visible?

File: infra/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+| `SSH_TARGET_IP`  |  Mgr node IP  |
+| `SSH_USER`  | Mgr username for ssh   |
+| `WIREGUARD_ENDPOINT`  | IP and port of the wireguard server for deployment in the format `<IP>:<Port>`   |
+| `WIREGUARD_OVERLAY_NETWORK_IP`  | Overlay network IP for wireguard server used for deployment   |
+| `WIREGUARD_PEER_PUBLIC_KEY`  | Public key of the wireguard server for deployment   |
+| `WIREGUARD_PRIVATE_KEY`  |  Private key for connecting to wireguard for deployment  |
+
+3. Create a new environment specific deployment workflow similar to `.github/workflows/deploy_prod1.yaml`
+
+4. Set variables in `values.yaml` and `secret.values.yaml` on the manager server in `/root/` (one directory above `PROJECT_PATH`)
```
Comment (WillNilges) [inline infra/README.md:25]: "Manager Server"? Which node is this? LB, or the K8S mgr?

File: infra/helm/meshdb/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+description: A Helm chart for Kubernetes
+
+# A chart can be either an 'application' or a 'library' chart.
+#
+# Application charts are a collection of templates that can be packaged into versioned archives
+# to be deployed.
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
```
Comment (WillNilges) [inline infra/helm/meshdb/Chart.yaml:12]: Nit: Delete all these comments?

File: infra/helm/meshdb/README.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+| autoscaling.enabled | bool | `false` |  |
+| autoscaling.maxReplicas | int | `100` |  |
+| autoscaling.minReplicas | int | `1` |  |
+| autoscaling.targetCPUUtilizationPercentage | int | `80` |  |
+| aws.access_key_id | string | `"the_real_value"` | AWS access key id for S3 |
+| aws.secret_access_key | string | `"the_real_value"` | AWS secret access key for S3 |
+| fullnameOverride | string | `"meshdb"` | App name |
+| image.pullPolicy | string | `"Always"` | pullPolicy for all images, should be `Always` |
+| map.base_url | string | `"http://admin-map.grandsvc.mesh.nycmesh.net"` | Map url |
+| meshdb_app_namespace | string | `"meshdbdev0"` | K8s namespace used for all resources |
```
Comment (WillNilges) [inline infra/helm/meshdb/README.md:18]: You could probably just make this `meshdb` everywhere, on both dev and prod.

File: infra/helm/meshdb/README.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+| meshweb.affinity | object | `{}` |  |
+| meshweb.backup_s3_base_folder | string | `"meshdb-backups/development/"` | Base folder for django postgres backups |
+| meshweb.backup_s3_bucket_name | string | `"meshdb-data-backups"` | Bucket used for django postgres backups |
+| meshweb.disable_pano_edits | string | `"True"` | Feature flag for disabling panorama edits |
+| meshweb.disable_profiling | string | `"True"` | Disable profiling in meshweb |
+| meshweb.django_secret_key | string | `"the_real_value"` | Django secret key |
```
Comment (WillNilges) [inline infra/helm/meshdb/README.md:24]: Nit: Point out that you can see the instructions in the README

File: infra/helm/meshdb/example.values.yaml [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+#serviceAccount:
+#  # Specifies whether a service account should be created
+#  create: true
+#  # Automatically mount a ServiceAccount's API credentials?
+#  automount: true
+#  # Annotations to add to the service account
+#  annotations: {}
+#  # The name of the service account to use.
+#  # If not set and create is true, a name is generated using the fullname template
+#  name: ""
```
Comment (WillNilges) [inline infra/helm/meshdb/example.values.yaml:109]: Delete comment?

File: infra/helm/meshdb/example.values.yaml [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+  # We usually recommend not to specify default resources and to leave this as a conscious
+  # choice for the user. This also increases chances charts run on environments with little
+  # resources, such as Minikube. If you do want to specify resources, uncomment the following
+  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  # limits:
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
```
Comment (WillNilges) [inline infra/helm/meshdb/example.values.yaml:33]: Delete comment? Maybe we should talk about this. Our resources are going to be so dummy small. Limiting to like 500mcore and 500mb of RAM is probably fine.

File: infra/helm/meshdb/templates/ingress.yaml
Comment (WillNilges) [inline infra/helm/meshdb/templates/ingress.yaml:1]: Delet?

File: infra/README.md (hunk 2 above)
Comment (WillNilges) [reply infra/README.md:25]: From Slack: Some of this is out of date.

Comment (WillNilges) [reply infra/README.md:25]: Also, kube mgrs will always have the correct k8s config.

**Review by WillNilges — APPROVED** [review]
I have a few nits

File: infra/helm/meshdb/templates/ingress.yaml
Comment (james-otten) [reply infra/helm/meshdb/templates/ingress.yaml:1]: Gone

File: .github/workflows/deploy_dev3.yaml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Install SSH key
+        uses: shimataro/ssh-key-action@d4fffb50872869abe2d9a9098a6d9c5aa7d16be4 # v2
+        with:
+          key: ${{ secrets.SSH_PRIVATE_KEY }}
+          name: id_ed25519 # optional
+          known_hosts: ${{ secrets.SSH_KNOWN_HOSTS }}
+          #config: ${{ secrets.CONFIG }} # ssh_config; optional
```
[the remark this answers is not in the document]
Comment (james-otten) [reply .github/workflows/deploy_dev3.yaml]: Gone

File: .github/workflows/deploy_dev3.yaml [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+on:
+  push:
+    branches: [ main, james/infra_final ]
+
+permissions: read-all
+
+jobs:
+    deploy_to_prod1:
+      name: Deploy to dev3
```
[the remark this answers is not in the document]
Comment (james-otten) [reply .github/workflows/deploy_dev3.yaml]: Done

File: infra/README.md (hunk 2 above)
Comment (james-otten) [reply infra/README.md]: Fixed up docs, line now gone

File: infra/helm/meshdb/example.values.yaml (hunk 6 above)
Comment (james-otten) [reply infra/helm/meshdb/example.values.yaml]: Entire file

File: infra/helm/meshdb/README.md (hunk 5 above)
Comment (james-otten) [reply infra/helm/meshdb/README.md:24]: Good call, added comment at top of file

File: infra/helm/meshdb/README.md (hunk 4 above)
Comment (james-otten) [reply infra/helm/meshdb/README.md]: updated

File: .github/workflows/helm_lint.yaml (hunk 1 above)
Comment (james-otten) [reply .github/workflows/helm_lint.yaml:35]: Yes, actual run is in the next step, this is just "do we even need to run it"

File: infra/helm/meshdb/Chart.yaml (hunk 3 above)
Comment (james-otten) [reply infra/helm/meshdb/Chart.yaml]: Gone

File: infra/helm/meshdb/example.values.yaml (hunk 7 above)
Comment (james-otten) [reply infra/helm/meshdb/example.values.yaml]: Entire file gone

**Pull request merged**